core/src/main/java/com/flipkart/varadhi/core/cluster/ConsumerApi.java

```diff
@@ -21,5 +19,7 @@ public interface ConsumerApi {
 
     CompletableFuture<ConsumerInfo> getConsumerInfo();
 
-    CompletableFuture<List<Message>> getMessages(GetMessagesRequest messagesRequest);
+    CompletableFuture<ShardDlqMessageResponse> getMessagesByTimestamp(long earliestFailedAt, int max_limit);
+
+    CompletableFuture<ShardDlqMessageResponse> getMessagesByOffset(String pageMarkers, int max_limit);
```

Comment:
String pageMarkers: "String" type looks too generic.
Why is it a string and how this string should be interpreted?

Reply:
ok. got it. the pageMarkers are "abstract" even for the web-api. These are returned by the consumers as part of the ShardDLQMessageResponse.

So, consumer come up with this value, and they can understand them. for the other components, it is just a abstract string.

Reply:
yes, that's the intent.